### PR TITLE
Remove matscipy dependency

### PR DESCRIPTION
**Pull request by SebastianM-C**
The `neighbourlist` tests were moved to the MolSimPy package and that package was added as a conditional dependency with Requires.jl as discussed in #13

### CODE REVIEW

File: src/nlist.jl [hunk 1]
```diff
@@ -4,18 +4,9 @@
...
-   # if !has_transient(at, (:nlist, cutoff)) || recompute
-   #    # this nlist will be destroyed as soon as positions change
-   #    set_transient!(at, (:nlist, cutoff), MatSciPy.NeighbourList(at, cutoff))
-   # end
-   # return get_transient(at, (:nlist, cutoff))
-   return MatSciPy.NeighbourList(at, cutoff)
-end
+# include("MatSciPy.jl")
+using MolSimPy
+using .MatSciPy: neighbourlist
```
Comment (cortner) [inline src/nlist.jl]: why is this line still here? Should this not import from `MolSimPy`

**Comment by cortner** [comment]
Is there a way to have an if-else version of `@require` so that a warning or error message can be printed when somebody tries to load the ASE neighbourlist?

File: src/nlist.jl (hunk 1 above)
Comment (SebastianM-C) [reply src/nlist.jl]: yeah, sorry.

**Comment by SebastianM-C** [comment]
I'm not sure. I'll look into it.

**Comment by SebastianM-C** [comment]
I'm not sure it's possible given https://github.com/MikeInnes/Requires.jl/issues/41

**Comment by cortner** [comment]
Ok, then if  I can indulge in your patience a little longer, could you try something the following: 
```
function neighbourlist(at::ASEAtoms, cutoff::Float64)::MatSciPy.NeighbourList
   if ASE.hasmatscipy
      return MatSciPy.NeighbourList(at, cutoff)
   else 
      @warn("some warning / error message") 
      return nothing 
   end 
end
```
And  the `@require` statement needs to set the flag. I think that may have to be done via `Ref` if I recall correctly.

**Comment by SebastianM-C** [comment]
That's a nice idea. I'll try to do it today / tonight. Sorry for the delay.

**Comment by SebastianM-C** [comment]
I moved the `neighbourlist` function to ASE as you suggested and removed it from MolSimPy (to avoid conflicts when both are loaded); See also the last two commits here: https://github.com/JuliaMolSim/MolSimPy.jl/commits/master

I have tested this (ASE.jl) locally on windows and it works (i.e. test pass). If you would like to extend the CI to windows (or use the previously mentioned github actions workflows), let me know.

**Comment by SebastianM-C** [comment]
I think what remains to be done is to add CI to MolSimPy.jl (see https://github.com/JuliaMolSim/MolSimPy.jl/issues/1)

**Comment by cortner** [comment]
Many thanks - I’ll review and merge tomorrow!

**Comment by cortner** [comment]
Very elegant. THank you!

**Comment by cortner** [comment]
I ll register a new version later, but I’ll first want to test my other packages

**Pull request merged**